sql/report_queries/HathiTrust reports/Serials Holdings Report/serials_holdings.sql

```diff
@@ -0,0 +1,139 @@
+		Item hrid
+	Item Statistical Codes Folio Reporting table:
+		Item statistical type code name
+		Item statistical code name 
+	Holdings Statistical Codes Folio Reporting table:
+		Holdings statistical type code name
+		Holdings statistical code name 
+	Instance Statistical Codes Folio Reporting table:
+		Instance statistical type code name
+		Instance statistical code name 
+	Instance Identifiers Folio Reporting table:
+		Identifier type name
+	Locations_libraries Folio Reporting table:
+		Library name
+		Campus name
+		Institution name
+*/
+WITH parameters AS (
+	SELECT
+	 ---- Fill out one, leave others blank to filter location name or code ----
+        '' ::VARCHAR AS institution_filter, -- 'KÃ¸benhavns Universitet','Montoya College'
+        '' ::VARCHAR AS campus_filter, -- 'Main Campus','City Campus','Online'
+        '' ::VARCHAR AS library_filter, -- 'Datalogisk Institut','Adelaide Library'
+        '' ::VARCHAR AS location_filter, -- 'Main Library','Annex','Online'
+        '' ::VARCHAR AS mode_of_issuance_filter, -- 'serial'
+        '' ::VARCHAR AS status_name_filter, -- 'Cataloged'
+        '' ::VARCHAR AS identifier_filter -- 'ISSN'
+),
+location_filtering AS (
+	SELECT
+		ii.id AS i_id,
+		loc.location_id,
+		loc.location_name,
+		loc.campus_id,
+		loc.campus_name,
+		loc.library_id,
+		loc.library_name,
+		loc.institution_id,
+		loc.institution_name
+	FROM 
+		inventory_items AS ii
+		LEFT JOIN folio_reporting.locations_libraries AS loc ON ii.effective_location_id = loc.location_id 
+	WHERE 
+		loc.institution_name = (SELECT institution_filter FROM parameters)
+        OR loc.campus_name = (SELECT campus_filter FROM parameters)
+        OR loc.library_name = (SELECT library_filter FROM parameters)
+        OR loc.location_name = (SELECT location_filter FROM parameters)
+),
+oclc_no AS (
+	SELECT
+		ii2.instance_id,
+		ii2.identifier_type_name AS oclc_type,
+		ii2.identifier AS oclc_number
+	FROM 
+		folio_reporting.instance_identifiers AS ii2
+	WHERE 	
+		ii2.identifier_type_name = 'OCLC'
+),
+statistical_codes AS (
+	SELECT
+		ie.item_id,
+		isc.statistical_code_type_name AS item_stat_code_type_name,
+		isc.statistical_code_name AS item_stat_code_name,
+		he.holdings_id,
+		hsc.statistical_code_type_name AS hol_stat_code_type_name,
+		hsc.statistical_code_name AS hol_stat_code_name,
+		ie2.instance_id,
+		isc3.statistical_code_type_name AS inst_stat_code_type_name,
+		isc3.statistical_code_name AS inst__stat_code_name
+	FROM 
+		folio_reporting.item_ext AS ie 
+		RIGHT JOIN folio_reporting.item_statistical_codes AS isc ON ie.item_id = isc.item_id
+		RIGHT JOIN folio_reporting.holdings_ext AS he ON ie.holdings_record_id = he.holdings_id 
+		RIGHT JOIN folio_reporting.instance_ext AS ie2 ON he.instance_id = ie2.instance_id 
+		RIGHT JOIN folio_reporting.holdings_statistical_codes AS hsc ON  he.holdings_id = hsc.holdings_id 
+		RIGHT JOIN folio_reporting.instance_statistical_codes AS isc3 ON ie2.instance_id = isc3.instance_id
```

Comment:
I'm wondering why `RIGHT JOIN` is used here when before 'LEFT JOIN' is used in the stat codes subqueries.  
Keep in mind that joining right will only give results with statistical codes set on every level (item, holding and instance).

Reply:
This last HathiTrust was tricky. The right join was the only way that I could get the distinct title list. Otherwise, there was duplication which isn't needed.

Reply:
Sure, when it works as it is intended, it's fine. I was just wondering. :)